### ISSUE TITLE
Calling "select" programmatically does not trigger "selectOverlap" callback.  Thus, I cannot "block" event overlap selection when this is done programmatically.

### ISSUE THREAD

**Issue by bs-thomas** [fullcalendar/fullcalendar#4150]
**Problem**
Calling "select" programmatically does not trigger "selectOverlap" callback.  Thus, I cannot "block" event overlap selection when this is done programmatically.  However, highlight selecting using mouse works.

**Expected Behavior**
Regardless of what method I use to select dates, the callback for "selectOverlap" should always trigger so we can capture this, and do customizations via this route.

**Codepen**
https://codepen.io/bs-thomas/pen/dmBWJY

**Comment by bs-thomas** [comment]
I also noticed that the 'select' method:
- ignores disabled dates
- does not trigger selectAllow callback

Thanks!

**Comment by acerix** [comment]
I believe this is intentional, eg. documentation for `selectOverlap` says it determines when "the user will not be allowed to select periods", ie. it only applies to selections by the user.

You could define the function for `selectOverlap` outside the calendar, and use the same function inside `select` to see if it's allowed.  

Or are you looking for a new feature to have the user-selection rules also apply to `select` method.

**Comment by arshaw** [comment]
i agree that selectOverlap and selectAllow should apply to programmatic selection. i'll consider this change for future versions

**Comment by nagyalex** [comment]
@bs-thomas Did you manage to find a temporary solution till this gets supported officially?

**Comment by a-zhunusbekov** [comment]
> I believe this is intentional, eg. documentation for `selectOverlap` says it determines when "the user will not be allowed to select periods", ie. it only applies to selections by the user.
> 
> You could define the function for `selectOverlap` outside the calendar, and use the same function inside `select` to see if it's allowed.
> 
> Or are you looking for a new feature to have the user-selection rules also apply to `select` method.

Hello! 

The problem still existing, so can I "_You could define the function for selectOverlap outside the calendar, and use the same function inside select to see if it's allowed._"?

**Comment by a-zhunusbekov** [comment]
> I also noticed that the 'select' method:
> 
> * ignores disabled dates
> * does not trigger selectAllow callback
> 
> Thanks!

Hello! @bs-thomas , Did you find or write any solution for that problem?
